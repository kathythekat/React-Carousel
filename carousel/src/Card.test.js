import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";

it("renders without crashing", function () {
  render(<Card caption="Photo by Richard Pasquarella on Unsplash" src="image1" />);
});
// end

it("has the correct alt text & src", function () {
  const { container, debug } = render(<Card caption="Photo by Richard Pasquarella on Unsplash" src="image1" />);
  const img = container.querySelector("img");
  debug(img);

  expect(img.getAttribute("alt")).toEqual("Photo by Richard Pasquarella on Unsplash");
  expect(img.getAttribute("src")).toContain("image1");
});
// end

it("matches snapshot", function () {
  const { container } = render(<Card caption="Photo by Richard Pasquarella on Unsplash" src="image1" />);
  expect(container).toMatchSnapshot();
});
// end
