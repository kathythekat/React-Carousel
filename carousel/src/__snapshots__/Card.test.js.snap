// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      Photo by Richard Pasquarella on Unsplash
    </h4>
    <img
      alt="Photo by Richard Pasquarella on Unsplash"
      class="Card-image"
      src="image1"
    />
    <small
      class="Card-small"
    >
      Image 
       of 
      .
    </small>
  </div>
</div>
`;
